packages/block-editor/src/store/actions.js

```diff
@@ -0,0 +1,517 @@
+ *
+ * @return {Object} Action object.
+ */
+export function moveBlockToPosition( clientId, fromRootClientId, toRootClientId, index ) {
+	return {
+		type: 'MOVE_BLOCK_TO_POSITION',
+		fromRootClientId,
+		toRootClientId,
+		clientId,
+		index,
+	};
+}
+
+/**
+ * Returns an action object used in signalling that a single block should be
+ * inserted, optionally at a specific index respective a root block list.
+ *
+ * @param {Object}  block            Block object to insert.
+ * @param {?number} index            Index at which block should be inserted.
+ * @param {?string} rootClientId     Optional root client ID of block list on which to insert.
+ * @param {?boolean} updateSelection If true block selection will be updated. If false, block selection will not change. Defaults to true.
+ *
+ * @return {Object} Action object.
+ */
+export function insertBlock( block, index, rootClientId, updateSelection = true ) {
+	return insertBlocks( [ block ], index, rootClientId, updateSelection );
+}
+
+/**
+ * Returns an action object used in signalling that an array of blocks should
+ * be inserted, optionally at a specific index respective a root block list.
+ *
+ * @param {Object[]} blocks          Block objects to insert.
+ * @param {?number}  index           Index at which block should be inserted.
+ * @param {?string}  rootClientId    Optional root client ID of block list on which to insert.
+ * @param {?boolean} updateSelection If true block selection will be updated.  If false, block selection will not change. Defaults to true.
+ *
+ * @return {Object} Action object.
+ */
+export function insertBlocks( blocks, index, rootClientId, updateSelection = true ) {
+	return {
+		type: 'INSERT_BLOCKS',
+		blocks: castArray( blocks ),
+		index,
+		rootClientId,
+		time: Date.now(),
+		updateSelection,
+	};
+}
+
+/**
+ * Returns an action object used in signalling that the insertion point should
+ * be shown.
+ *
+ * @param {?string} rootClientId Optional root client ID of block list on
+ *                               which to insert.
+ * @param {?number} index        Index at which block should be inserted.
+ *
+ * @return {Object} Action object.
+ */
+export function showInsertionPoint( rootClientId, index ) {
+	return {
+		type: 'SHOW_INSERTION_POINT',
+		rootClientId,
+		index,
+	};
+}
+
+/**
+ * Returns an action object hiding the insertion point.
+ *
+ * @return {Object} Action object.
+ */
+export function hideInsertionPoint() {
+	return {
+		type: 'HIDE_INSERTION_POINT',
+	};
+}
+
+/**
+ * Returns an action object resetting the template validity.
+ *
+ * @param {boolean}  isValid  template validity flag.
+ *
+ * @return {Object} Action object.
+ */
+export function setTemplateValidity( isValid ) {
+	return {
+		type: 'SET_TEMPLATE_VALIDITY',
+		isValid,
+	};
+}
+
+/**
+ * Returns an action object synchronize the template with the list of blocks
+ *
+ * @return {Object} Action object.
+ */
+export function synchronizeTemplate() {
```

Comment:
Trying to recall, but I think this is another "undesirable" action we oughtn't need. I'll try to dig up a reference.